examples/additional-rpc-namespace-in-cli/src/main.rs

```diff
@@ -14,24 +14,39 @@
 
 use clap::Parser;
 use jsonrpsee::{core::RpcResult, proc_macros::rpc};
-use reth::cli::{
-    components::{RethNodeComponents, RethRpcComponents},
-    config::RethRpcConfig,
-    ext::{RethCliExt, RethNodeCommandConfig},
-    Cli,
-};
+use reth::cli::Cli;
+use reth_node_ethereum::EthereumNode;
 use reth_transaction_pool::TransactionPool;
 
 fn main() {
-    Cli::<MyRethCliExt>::parse().run().unwrap();
-}
-
-/// The type that tells the reth CLI what extensions to use
-struct MyRethCliExt;
-
-impl RethCliExt for MyRethCliExt {
-    /// This tells the reth CLI to install the `txpool` rpc namespace via `RethCliTxpoolExt`
-    type Node = RethCliTxpoolExt;
+    Cli::<RethCliTxpoolExt>::parse()
+        .run(|builder, args| async move {
+            let handle = builder
+                .with_types(EthereumNode::default())
+                .with_components(EthereumNode::components())
+                .extend_rpc_modules(move |ctx| {
```

Comment:
I assume people will want to extend these with functions vs doing it all in-line, so maybe we should refactor the closure logic to a helper function to make this more obvious i.e.

```
.extend_rpc_modules(move |ctx| extend_txpool(&mut ctx)?; println!("extensions configured"); Ok(()) })

Reply:
I think this is fine, but can add more examples later